Error with mingw-w64

I get an error compiling the package using mingw-w64 on windows.  With netcdf-4.3.3, I get the same error when using CMake to configure the build, but not when I use `./configure`.


```
git clone https://github.com/Unidata/netcdf-c.git && cd netcdf-c
mkdir build && cd build
cmake -G "MSYS Makefiles" -DCMAKE_BUILD_TYPE=Release ..
make
```

```
# NetCDF C Configuration Summary
==============================

# General
-------
NetCDF Version:         4.4.0-development
Configured On:          Tue, Dec  8, 2015  1:42:02 AM
Host System:            x86_64-MINGW64_NT-6.3-2.4.0(0.292/5/3)
Build Directory:        C:/msys64/home/Administrator/netcdf-c/build
Install Prefix:         C:/Program Files (x86)/netCDF

# Compiling Options
-----------------
C Compiler:             C:/msys64/mingw64/bin/gcc.exe
CFLAGS:                  -O3 -DNDEBUG
CPPFLAGS:
LDFLAGS:
AM_CFLAGS:
AM_CPPFLAGS:
AM_LDFLAGS:
Shared Library:         yes
Static Library:         no
Extra libraries:        -lhdf5_hl -lhdf5 -ldl -lz -lszip -lcurl

# Features
--------
NetCDF-2 API:           yes
HDF4 Support:           no
NetCDF-4 API:           yes
NC-4 Parallel Support:  no
PNetCDF Support:        no
DAP Support:            yes
Diskless Support:       yes
MMap Support:           no
JNA Support:            no

[ ... snip ...]

Scanning dependencies of target netcdf3
[  9%] Building C object libsrc/CMakeFiles/netcdf3.dir/v1hpg.c.obj
[  9%] Building C object libsrc/CMakeFiles/netcdf3.dir/putget.c.obj
C:/msys64/home/Administrator/netcdf-c/libsrc/putget.c: In function 'NC3_get_vara':
C:/msys64/home/Administrator/netcdf-c/libsrc/putget.c:10397:5: warning: implicit declaration of function 'ALLOC_ONSTACK' [-Wimplicit-function-declaration]
     ALLOC_ONSTACK(coord, size_t, varp->ndims);
     ^
C:/msys64/home/Administrator/netcdf-c/libsrc/putget.c:10397:19: error: 'coord' undeclared (first use in this function)
     ALLOC_ONSTACK(coord, size_t, varp->ndims);
                   ^
C:/msys64/home/Administrator/netcdf-c/libsrc/putget.c:10397:19: note: each undeclared identifier is reported only once for each function it appears in
C:/msys64/home/Administrator/netcdf-c/libsrc/putget.c:10397:26: error: expected expression before 'size_t'
     ALLOC_ONSTACK(coord, size_t, varp->ndims);
                          ^
C:/msys64/home/Administrator/netcdf-c/libsrc/putget.c:10398:19: error: 'upper' undeclared (first use in this function)
     ALLOC_ONSTACK(upper, size_t, varp->ndims);
                   ^
C:/msys64/home/Administrator/netcdf-c/libsrc/putget.c:10398:26: error: expected expression before 'size_t'
     ALLOC_ONSTACK(upper, size_t, varp->ndims);
                          ^
C:/msys64/home/Administrator/netcdf-c/libsrc/putget.c:10427:5: warning: implicit declaration of function 'FREE_ONSTACK' [-Wimplicit-function-declaration]
     FREE_ONSTACK(upper);
     ^
C:/msys64/home/Administrator/netcdf-c/libsrc/putget.c: In function 'NC3_put_vara':
C:/msys64/home/Administrator/netcdf-c/libsrc/putget.c:10530:19: error: 'coord' undeclared (first use in this function)
     ALLOC_ONSTACK(coord, size_t, varp->ndims);
                   ^
C:/msys64/home/Administrator/netcdf-c/libsrc/putget.c:10530:26: error: expected expression before 'size_t'
     ALLOC_ONSTACK(coord, size_t, varp->ndims);
                          ^
C:/msys64/home/Administrator/netcdf-c/libsrc/putget.c:10531:19: error: 'upper' undeclared (first use in this function)
     ALLOC_ONSTACK(upper, size_t, varp->ndims);
                   ^
C:/msys64/home/Administrator/netcdf-c/libsrc/putget.c:10531:26: error: expected expression before 'size_t'
     ALLOC_ONSTACK(upper, size_t, varp->ndims);
                          ^
libsrc/CMakeFiles/netcdf3.dir/build.make:98: recipe for target 'libsrc/CMakeFiles/netcdf3.dir/putget.c.obj' failed
make[2]: *** [libsrc/CMakeFiles/netcdf3.dir/putget.c.obj] Error 1
CMakeFiles/Makefile2:1253: recipe for target 'libsrc/CMakeFiles/netcdf3.dir/all' failed
make[1]: *** [libsrc/CMakeFiles/netcdf3.dir/all] Error 2
Makefile:160: recipe for target 'all' failed
make: *** [all] Error 2
```

--- a/include/onstack.h
+++ b/include/onstack.h
@@ -11,15 +11,15 @@
  * "allocate" arrays on the stack where possible.
  * (Where not possible, malloc and free are used.)
  *
- * The macro ALLOC_ONSTACK(name, type, nelems) is used to declare 
+ * The macro ALLOC_ONSTACK(name, type, nelems) is used to declare
  * an array of 'type' named 'name' which is 'nelems' long.
  * FREE_ONSTACK(name) is placed at the end of the scope of 'name'
  * to call 'free' if necessary.
- * 
+ *
  * The macro ALLOC_ONSTACK wraps a call to alloca() on most systems.
  */
 
-#ifdef _MSC_VER 
+#ifdef _WIN32
 #ifdef HAVE_MALLOC_H
 #undef HAVE_ALLOCA
 #define HAVE_ALLOCA 1
@@ -80,5 +80,5 @@
 	free(name)
 
 #endif
-	
+
 #endif /* _ONSTACK_H_ */

--- a/oc2/ocinternal.c
+++ b/oc2/ocinternal.c
@@ -12,7 +12,7 @@
 #endif
 #include <errno.h>
 
-#ifdef _WIN32
+#ifdef _MSC_VER
 typedef int pid_t;
 #endif
 

